src/sheet/sheet.c

```diff
@@ -317,6 +316,10 @@ gboolean sheet_get_adjustments (const Sheet *sheet, GtkAdjustment **hadj, GtkAdj
  */
 void sheet_change_zoom (Sheet *sheet, gdouble factor)
 {
+	double zoom;
+	sheet_get_zoom (sheet, &zoom);
+	sheet_set_zoom (sheet, zoom * factor);
+
```

Comment:
No, sheet_set_zoom() no longer calls goo_canvas_set_scale(). That was
wrong and has been changed !

Also, sheet_change_zoom() is not the appropriate function to use to reset the zoom in a new sheet, because such function is used for zooming in and out by one zoom step only.

I believe the zoom code is fine as it is.

Please test it again to confirm this.